src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -0,0 +1,583 @@
+from azure.cli.core._profile import Profile
+from azext_connectedk8s._client_factory import _graph_client_factory
+from azext_connectedk8s._client_factory import cf_resource_groups
+from azext_connectedk8s._client_factory import _resource_client_factory
+from msrestazure.azure_exceptions import CloudError
+from kubernetes import client as kube_client, config, watch  # pylint: disable=import-error
+from Crypto.IO import PEM  # pylint: disable=import-error
+from Crypto.PublicKey import RSA  # pylint: disable=import-error
+from Crypto.Util import asn1  # pylint: disable=import-error
+
+from .vendored_sdks.models import ConnectedCluster, ConnectedClusterAADProfile, ConnectedClusterIdentity
+
+
+logger = get_logger(__name__)
+
+
+# pylint:disable=unused-argument
+# pylint: disable=too-many-locals
+# pylint: disable=too-many-branches
+# pylint: disable=too-many-statements
+# pylint: disable=line-too-long
+def create_connectedk8s(cmd, client, resource_group_name, cluster_name, location=None,
+                        kube_config=None, kube_context=None, no_wait=False, tags=None):
+    logger.warning("Ensure that you have the latest helm version installed before proceeding.")
+    logger.warning("This operation might take a while...\n")
+
+    # Setting subscription id
+    subscription_id = get_subscription_id(cmd.cli_ctx)
+
+    # Setting user profile
+    profile = Profile(cli_ctx=cmd.cli_ctx)
+
+    # Fetching Tenant Id
+    graph_client = _graph_client_factory(cmd.cli_ctx)
+    onboarding_tenant_id = graph_client.config.tenant_id
+
+    # Setting kubeconfig
+    kube_config = set_kube_config(kube_config)
+
+    # Removing quotes from kubeconfig path. This is necessary for windows OS.
+    trim_kube_config(kube_config)
+
+    # Loading the kubeconfig file in kubernetes client configuration
+    try:
+        config.load_kube_config(config_file=kube_config, context=kube_context)
+    except Exception as e:
+        raise CLIError("Problem loading the kubeconfig file." + str(e))
+    configuration = kube_client.Configuration()
+
+    # Checking the connection to kubernetes cluster.
+    # This check was added to avoid large timeouts when connecting to AAD Enabled AKS clusters
+    # if the user had not logged in.
+    check_kube_connection(configuration)
+
+    # Checking helm installation
+    check_helm_install(kube_config, kube_context)
+
+    # Check helm version
+    check_helm_version(kube_config, kube_context)
+
+    # Validate location
+    resourceClient = _resource_client_factory(cmd.cli_ctx, subscription_id=subscription_id)
+    if location is None:
+        try:
+            location = resourceClient.resource_groups.get(resource_group_name).location
+        except:
+            raise CLIError("The resource group '{}' does not exist and has to be created. "
+                           .format(resource_group_name) +
+                           "Use the --location flag to specify the location to create the resource group.")
```

Comment:
This is getting location of the resource group passed in, right? We have a validator for it. You can add the following code in `_params.py`:
https://github.com/Azure/azure-cli-extensions/blob/0f9aa5840b130561c5b71033f2b06972de504acd/src/databricks/azext_databricks/_params.py#L23